src/core/pages/PatientSummary/DialogWithStyles.js

```diff
@@ -0,0 +1,158 @@
+     *
+     * @author Bogdan Shcherban <bsc@piogroup.net>
+     * @param {string} itemName
+     */
+    selectHeading = itemName => {
+        const { selectedHeadings } = this.state;
+        let headingsArray = selectedHeadings;
+        if (this.isHeadingChecked(itemName)) {
+            let index = headingsArray.indexOf(itemName);
+            headingsArray.splice(index, 1)
+        } else {
+            headingsArray.push(itemName);
+        }
+        this.setState({
+            selectedHeadings: headingsArray,
+        });
+        this.props.setHeadingsAction(headingsArray);
+    };
+
+    /**
+     * This function toggle shows mode
+     *
+     * @author Bogdan Shcherban <bsc@piogroup.net>
+     * @param {string} mode
+     */
+    selectShowMode = mode => {
+        this.setState({
+            selectedMode: mode,
+        });
+        this.props.setModeAction(mode);
+    };
+
+    render() {
+        const { classes, onClose, showMode, ...rest } = this.props;
+        const { selectedMode, selectedHeadings } = this.state;
+        return (
+            <Dialog onBackdropClick={() => onClose()} {...rest}>
+                <div className={classes.dialogBlock} >
+                    <Typography>SHOW</Typography>
+                    <Divider />
+                    <div className={classes.dialogItem}>
+                        {
+                            synopsisData.map(item => {
+                                return (
+                                    <div className={classes.dialogLabel}>
+                                        <Checkbox checked={this.isHeadingChecked(item.list)} color="primary" onChange={() => this.selectHeading(item.list)} />
+                                        <span>{item.title}</span>
+                                    </div>
+                                );
+                            })
+                        }
+                    </div>
+                    <Typography>FEEDS</Typography>
+                    <Divider />
+                    <div className={classes.dialogItem}></div>
+                    <Typography>VIEW OF BOARDS</Typography>
+                    <Divider />
+                    <div className={classes.dialogItemColumn}>
+                    {
+                        showModesArray.map(item => {
+                            return (
+                                <label className={classes.dialogLabel}>
+                                    <Radio checked={selectedMode === item.type} color="primary" onChange={() => this.selectShowMode(item.type)} />
+                                    <span>{item.label}</span>
+                                </label>
+                            );
+                        })
+                    }
+                    </div>
+                </div>
+            </Dialog>
+        );
+    }
+};
+
+const mapStateToProps = state => {
+    return {
+        showMode: get(state, 'custom.showMode.data', null),
```

Comment:
I don't understand why you use `get` here. Access the state directly.

Reply:
I removed get. 

I used it because { get } gives possibility to avoid error if required object won't be found.